lib/net/ntlm/encode_util.rb

```diff
@@ -27,7 +27,7 @@ def self.swap16(str)
       # Decode a UTF16 string to a ASCII string
       # @param [String] str The string to convert
       def self.decode_utf16le(str)
-        str.force_encoding(Encoding::UTF_16LE)
+        str.dup.force_encoding(Encoding::UTF_16LE)
```

Comment:
I tried to find a way to make this blow up if that first encoding was omitted to better catch this in the unit tests but ruby seems very forgiving at least on the strings and encoding I was using